utils.py

```
@@ -0,0 +1,194 @@
+        dict: A dictionary containing statistics for each variable in the output cube. Keys are variable names 
+              (matching the pattern 'B01, B02, ...') and values are dictionaries containing mean, variance, min, max, 
+              and quantile statistics.
+    """
+
+    statistics = {}
+    band_names = [band_name for band_name in hypercube.data_vars\
+                      if re.match(r'^B\d',band_name)]
+
+    for band_name in band_names:
+
+        band_data = hypercube[band_name]
+
+        mean_value = float(band_data.mean())
+        variance_value = float(band_data.var())
+        
+        min_value = float(band_data.min())
+        max_value = float(band_data.max())
+        
+        quantiles = band_data.quantile([0.25, 0.5, 0.75]).values
+        
+        statistics[band_name] = {
+            'mean': mean_value,
+            'variance': variance_value,
+            'min': min_value,
+            'max': max_value,
+            'quantile25': quantiles[0],
+            'quantile50': quantiles[1],
+            'quantile75': quantiles[2],
+        }
+
+    return statistics
+
+def assert_band_statistics(output_band_stats: dict, scenario_name: str, tolerance: float = 0.01) -> None:
+    """
+    Asserts the output band statistics against a reference dictionary stored in a JSON file with tolerance.
+
+    Parameters:
+        output_band_stats (dict): The output dictionary containing band statistics to be asserted.
+        reference_file (str): The path to the JSON file containing the reference dictionary.
+        tolerance (float): Tolerance value for comparing values. Defaults to 0.01.
+
+    Returns:
+        None
+    """
+    try:
+        groundtruth_band_stats = extract_reference_band_statistics(scenario_name)
+        compare_band_statistics(output_band_stats, groundtruth_band_stats, tolerance)
+    except FileNotFoundError:
+        print(f"Error: Reference file '{scenario_name}' not found.")
+    except json.JSONDecodeError:
```

I would not do these try-excepts that ignore error: if there is something is wrong this should fail hard I think